tests/at_functional_test/test/enroll_verb_test.dart

```diff
@@ -77,47 +79,364 @@ void main() {
     });
 
-    test('second enroll request using totp', () async {
-      await socket_writer(socketFirstAtsign!, 'from:$firstAtsign');
+    // Purpose of the tests
+    // 1. Do a pkam authentication
+    // 2. Send an enroll request
+    //  3 . Get an otp from the first client
+    //  4. Send an enroll request with otp from the second client
+    //  5. First client doesn't approve the enroll request
+    //  6. Second client should get an exception as the enroll request is not approved
+    test(
+        'second enroll request using totp and client did not approved enrollment request',
+        () async {
+      await socket_writer(socketConnection1!, 'from:$firstAtsign');
+      var fromResponse = await read();
+      print('from verb response : $fromResponse');
+      fromResponse = fromResponse.replaceAll('data:', '');
+      var pkamDigest = generatePKAMDigest(firstAtsign, fromResponse);
+      await socket_writer(socketConnection1!, 'pkam:$pkamDigest');
+      var pkamResult = await read();
+      expect(pkamResult, 'data:success\n');
+      var enrollRequest =
+          'enroll:request:appName:wavi:deviceName:pixel:namespaces:[wavi,rw]:apkamPublicKey:${pkamPublicKeyMap[firstAtsign]!}\n';
+      await socket_writer(socketConnection1!, enrollRequest);
+      var enrollResponse = await read();
+      print(enrollResponse);
+      enrollResponse = enrollResponse.replaceFirst('data:', '');
+      var enrollJsonMap = jsonDecode(enrollResponse);
+      expect(enrollJsonMap['enrollmentId'], isNotEmpty);
+      expect(enrollJsonMap['status'], 'success');
+
+      var totpRequest = 'totp:get\n';
+      await socket_writer(socketConnection1!, totpRequest);
+      var totpResponse = await read();
+      totpResponse = totpResponse.replaceFirst('data:', '');
+      print(totpResponse);
+      totpResponse = totpResponse.trim();
+
+      // connect to the second client
+      socketConnection2 =
+          await secure_socket_connection(firstAtsignServer, firstAtsignPort);
+      socket_listener(socketConnection2!);
+
+      //send second enroll request with totp
+      var apkamPublicKey = pkamPublicKeyMap[firstAtsign];
+      var secondEnrollRequest =
+          'enroll:request:appName:buzz:deviceName:pixel:namespaces:[buzz,rw]:totp:$totpResponse:apkamPublicKey:$apkamPublicKey\n';
+      print(secondEnrollRequest);
+      await socket_writer(socketConnection2!, secondEnrollRequest);
+
+      var secondEnrollResponse = await read();
+      print(secondEnrollResponse);
+      secondEnrollResponse = secondEnrollResponse.replaceFirst('data:', '');
+      var enrollJson = jsonDecode(secondEnrollResponse);
+      expect(enrollJson['enrollmentId'], isNotEmpty);
+      expect(enrollJson['status'], 'pending');
+
+      var secondEnrollId = enrollJson['enrollmentId'];
+
+      // deny the enroll request from the first client
+      var denyEnrollCommand = 'enroll:deny:enrollmentId:$secondEnrollId\n';
```

Comment:
instead of denying through ssl client..deny on receiving notification..you should call notification.subscribe(..).listen and then have the deny logic inside that.Check the sample gist code I shared